app/src/main/java/com/nowak/wjw/simplecompass/data/location/LocationApiHandler.java

```diff
@@ -27,44 +28,36 @@ public LocationApiHandler(FusedLocationProviderClient mFusedLocationProviderClie
     @Override
     public void onLocationResult(LocationResult locationResult) {
         super.onLocationResult(locationResult);
+//        Timber.d("onLocationResult %s %s", locationResult.getLastLocation().getLatitude(), locationResult.getLastLocation().getLongitude());
         mLocation.setValue(locationResult.getLastLocation());
     }
 
     public LiveData<Location> getLocation() {
         return mLocation;
     }
 
+
     /**
      * Call requires permission which may be rejected by user:
      * code should explicitly check to see if permission is available (with checkPermission) or explicitly handle a potential SecurityException
      *
      * @throws SecurityException
      */
-    public void getLastLocation(Activity activity) throws SecurityException {
+    public void initiateLastLocation() throws SecurityException {
         mFusedLocationClient.getLastLocation()
-                .addOnSuccessListener(activity, new OnSuccessListener<Location>() {
```

Comment:
addOnSuccessListener - to jest interfejs funkcyjny i nawet w javie masz tę możliwość że możesz tę klasę anonimową zastąpić lambdą. czyli
location -> {
//ciało metody onSuccess
}